### PR TITLE
Use fully qualified name for all macro non-prelude types

### PR DESCRIPTION
**Pull request by bgurney-rh**
Related: #522

### CODE REVIEW

File: src/units.rs [hunk 1]
```diff
@@ -2,11 +2,7 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-use std::{
-    fmt,
-    iter::Sum,
-    ops::{Add, AddAssign, Deref, Div, Mul, Rem, Sub, SubAssign},
-};
+use std::{fmt, iter::Sum};
```
Comment (mulkieran) [inline src/units.rs]: If you give Sum and Display the treatment in the range_marcros.rs macros all these std imports ought to go away (I think).

Comment (bgurney-rh) [reply src/units.rs]: Ah, that's interesting in how it's defined, so I see "impl fmt::Display" in the macro definitions.

I also see "impl fmt::Debug"; should that be qualified as well?

Comment (mulkieran) [reply src/units.rs]: Yes, it really should. I don't know why I'm surprised that Debug is not in the prelude, but evidently it isn't.

**Comment by bgurney-rh** [comment]
I added the qualifiers for id_macros.rs and shared_macros.rs.

File: src/range_macros.rs [hunk 2]
```diff
@@ -299,12 +299,7 @@ macro_rules! checked_add {
 #[cfg(test)]
 mod tests {
 
-    use std::{
-        fmt,
-        iter::Sum,
-        ops::{Add, AddAssign, Deref, Div, Mul, Rem, Sub, SubAssign},
-        u64,
-    };
+    use std::{fmt, u64};
```
Comment (mulkieran) [inline src/range_macros.rs]: I bet fmt use will go away when you make suggested changes.

Comment (bgurney-rh) [reply src/range_macros.rs]: Done; should be in the next force-push (after this most recent one; I forgot to run "cargo fmt").

**Comment by bgurney-rh** [comment]
That last series was a bit dizzying, so here's a diff of it just in case:
```
diff --git a/src/core/types.rs b/src/core/types.rs
index 725c1e0..8780999 100644
--- a/src/core/types.rs
+++ b/src/core/types.rs
@@ -9,7 +9,7 @@ use crate::{
         deviceinfo::{DM_NAME_LEN, DM_UUID_LEN},
         errors::ErrorKind,
     },
-    result::{DmError, DmResult},
+    result::DmError,
 };
 
 /// An error function to construct an error when creating a new string id.
diff --git a/src/id_macros.rs b/src/id_macros.rs
index 6741237..58ca076 100644
--- a/src/id_macros.rs
+++ b/src/id_macros.rs
@@ -35,13 +35,13 @@ macro_rules! str_check {
 macro_rules! str_id {
     ($B:ident, $O:ident, $MAX:ident, $err_func:ident) => {
         /// The borrowed version of the DM identifier.
-        #[derive(Debug, PartialEq, Eq, Hash)]
+        #[derive(std::fmt::Debug, PartialEq, Eq, std::hash::Hash)]
         pub struct $B {
             inner: str,
         }
 
         /// The owned version of the DM identifier.
-        #[derive(Debug, Clone, PartialEq, Eq, Hash)]
+        #[derive(std::fmt::Debug, Clone, PartialEq, Eq, std::hash::Hash)]
         pub struct $O {
             inner: String,
         }
@@ -49,7 +49,7 @@ macro_rules! str_id {
         impl $B {
             /// Create a new borrowed identifier from a `&str`.
             #[allow(clippy::new_ret_no_self)]
-            pub fn new(value: &str) -> DmResult<&$B> {
+            pub fn new(value: &str) -> $crate::result::DmResult<&$B> {
                 if let Some(err_msg) = str_check!(value, $MAX - 1) {
                     return Err($err_func(&err_msg));
                 }
@@ -72,7 +72,7 @@ macro_rules! str_id {
         }
 
         impl std::fmt::Display for $B {
-            fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
+            fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
                 write!(f, "{}", &self.inner)
             }
         }
@@ -80,7 +80,7 @@ macro_rules! str_id {
         impl $O {
             /// Construct a new owned identifier.
             #[allow(clippy::new_ret_no_self)]
-            pub fn new(value: String) -> DmResult<$O> {
+            pub fn new(value: String) -> $crate::result::DmResult<$O> {
                 if let Some(err_msg) = str_check!(&value, $MAX - 1) {
                     return Err($err_func(&err_msg));
                 }
@@ -111,11 +111,11 @@ macro_rules! str_id {
 
 #[cfg(test)]
 mod tests {
-    use std::{fmt, iter, ops::Deref};
+    use std::{iter, ops::Deref};
 
     use crate::{
         core::errors::{Error, ErrorKind},
-        result::{DmError, DmResult},
+        result::DmError,
     };
 
     fn err_func(err_msg: &str) -> DmError {
```

**Comment by bgurney-rh** [comment]
Okay, I think this is (hopefully) ready.

File: src/range_macros.rs (hunk 2 above)
Comment (bgurney-rh) [reply src/range_macros.rs]: It did.

**Pull request closed without merging**